### ISSUE TITLE
Rule MANIFEST - Extension type 'library' is not accepted by JED?

### ISSUE THREAD

**Issue by anibalsanchez** [joomla-extensions/jedchecker#136]
Hi,

Testing an extension, I found this false positive error.

Libraries are accepted on JED. It is a valid extension type.

CC @dryabov

**Comment by dryabov** [comment]
@anibalsanchez Just to be sure, what about others: `file`, `language`, `template`? I guess `language` and `template` are not accepted, but I'm not sure about `file`.

**Comment by anibalsanchez** [comment]
language and template are not accepted as standalone extensions.

In the extension where I detected the issue, the library is a part of a package. So, it is a completely accepted usage:

```
Rule MANIFEST - XML Manifests
1 Errors - Click to View Details
#001 /packages/lib_noboss/noboss.xml
Extension type 'library' is not accepted by JED
```

**Comment by dryabov** [comment]
I'm going to move this check to the XMLINFO ruleset, because it should be applied to the main manifest file only. But it requires PR #120 to be merged (as it introduces $isTopLevel parameter to distinguish main and dependent manifests).

@Llewellynvdm Could you try to merge PR #120?

**Comment by dryabov** [comment]
*Résumé*: PR#137 drops this check from XMLMANIFEST rules, and PR#120 inserts it to XMLINFO rules.

**Comment by ohrionmartin** [comment]
I tested this today on an extension with external libraries and it still gives the same error:
![Screenshot from 2021-07-10 14-41-16](https://user-images.githubusercontent.com/38046346/125163356-12624300-e18d-11eb-9caa-6284fb1ef30b.png)

**Comment by dryabov** [comment]
JED "officially" supports components, modules, and plugins only (they are the only options you have at the submission page). The TCPDF has type=library in the manifest, but submitted as an component (just because there is no other way). I was sure it's the only library on JED, but then found another one on https://extensions.joomla.org/category/core-enhancements/libraries/.

IMHO, the correct way would be to add library type to the extension types on JED (see https://github.com/joomla/com_jed4/blob/main/public_html/administrator/components/com_jed/forms/extension.xml#L68-L77), and modify TCPDF entry properly.

PS. @Llewellynvdm @anibalsanchez I've just found above-mentioned extension.xml file contains "component" and "module" values, but https://github.com/joomla/com_jed4/blob/main/public_html/administrator/components/com_jed/forms/filter_extensions.xml#L63-L75 contains "com" and "mod" instead. Most likely it is related to migration from JED3 to JED4 and not a problem actually, but to be sure check it.

**Comment by Llewellynvdm** [comment]
> I was sure it's the only library on JED

Wow how surprised I am [that I listed this](https://extensions.joomla.org/extension/core-enhancements/libraries/tcpdf-library/) like five or more years ago... I wonder if libraries was allowed back then... can't remember that far back. I have been updating that library here https://github.com/vdm-io/tcpdf but need to update it on the JED too.... anyway only today realized this, thanks @dryabov for pointing this issue out. Probably have to remove it then... will mention it to the JED team.

**Comment by anibalsanchez** [comment]
Please, check the terms of service for the definitive guide on these matters:

https://extensions.joomla.org/community/terms-of-service/

> d. The following will not be listed:
> i. Extension-specific listings for extensions not listed in the JED
> ii. Translations
> iii. Templates
> iv. Anything that doesn’t use the Joomla installer (excluding extension specific extensions)

**Comment by Llewellynvdm** [comment]
@dryabov have you corrected this issue?

**Comment by dryabov** [comment]
Yes, XMLInfo ruleset display an error message for library-type extensions.
